tests/unit/test_masks.py

```diff
@@ -153,6 +175,39 @@ def it_knows_its_mask_tile(self, request):
 
         assert binary_mask == expected_mask
 
+    def it_knows_its_mask_tile_and_compute_with_default(self, request):
+        tile = Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, None, None)
+
+        calculate_tissue_mask_tile_ = method_mock(
+            request, Tile, "calculate_tissue_mask"
+        )
+
+        tissue_mask = TissueMask()
+        _ = tissue_mask(tile)
+
+        calculate_tissue_mask_tile_.assert_called_once()
+
+    def it_knows_its_mask_tile_and_supports_custom_filters(self, request):
+        tile = Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, None, None)
+        custom_filter_cls = image_filters.RgbToGrayscale
+
+        custom_filters = [custom_filter_cls()]
+        custom_filter_call_ = method_mock(request, custom_filter_cls, "__call__")
+
+        set_custom_filters_ = method_mock(
+            request, _CustomFiltersComposition, "set_custom_filters"
+        )
+
+        # Call the tissue mask with custom filters
+        tissue_mask = TissueMask(*custom_filters)
+        _ = tissue_mask(tile)
+
+        # Ensure the custom tissue calculator, the internal setter
+        # and the custom filter itself is called
+        set_custom_filters_.assert_called_once()
+        assert set_custom_filters_.call_args[0][1] == custom_filters[0]
+        custom_filter_call_.assert_called_once()
```

Comment:
same

Reply:
Regarding the whole test: I feel that testing that `RgbToGrayscale` is called is not a good strategy, because:

1. usually we want to mock all the methods/instances that get called/used directly by the method we are testing, so in this case it would be `Tile.calculate_tissue_mask`
2. as a proof for my first point: this test is not really checking that the mask is built with the custom filters, because `RgbToGrayscale` is actually the first filter in the "default" filter sequence (for both `Slide` and Tile). So this test would actually pass with both the default and the "custom" filters represented by `[RgbToGrayscale()]` 🤓

Please let me know if this makes sense for you!

Reply:
@Christopher22 I think you misunderstood my points here. This updated test covers point number 2 but not number 1, the mocked object is not directly called/used by the method you're testing.